jenkins/internal-ci/centos-7.9.2009/pr/kubernetes/motr-k8-cicd.groovy

```diff
@@ -0,0 +1,335 @@
+					ln -s "${PYTHON_DEPS}" "${PYTHON_LIB_LOCATION}"
+				'''
+				sh label: 'RPM Signing', script: '''
+					pushd cortx-re/scripts/rpm-signing
+					cat gpgoptions >>  ~/.rpmmacros
+					sed -i 's/passphrase/'${PASSPHARASE}'/g' genkey-batch
+					gpg --batch --gen-key genkey-batch
+					gpg --export -a 'Seagate'  > RPM-GPG-KEY-Seagate
+					rpm --import RPM-GPG-KEY-Seagate
+					popd
+					pushd cortx-re/scripts/rpm-signing
+					chmod +x rpm-sign.sh
+					cp RPM-GPG-KEY-Seagate ${CORTX_ISO_LOCATION}
+					for rpm in `ls -1 ${CORTX_ISO_LOCATION}/*.rpm`
+					do
+						./rpm-sign.sh ${PASSPHARASE} ${rpm}
+					done
+					popd
+				'''
+				sh label: 'RPM Signing', script: '''
+					pushd ${CORTX_ISO_LOCATION}
+					rpm -qi createrepo || yum install -y createrepo
+					createrepo .
+					popd
+				'''	
+				sh label: 'RPM Signing', script: '''
+					pushd cortx-re/scripts/release_support
+					sh build_release_info.sh -v ${VERSION} -l ${CORTX_ISO_LOCATION} -t ${THIRD_PARTY_LOCATION}
+					sh build_readme.sh "${DESTINATION_RELEASE_LOCATION}"
+					popd
+					sed -i -e 's/BRANCH:.*/BRANCH: "motr-pr"/g' ${CORTX_ISO_LOCATION}/RELEASE.INFO
+					cp "${THIRD_PARTY_LOCATION}/THIRD_PARTY_RELEASE.INFO" "${DESTINATION_RELEASE_LOCATION}"
+					cp "${CORTX_ISO_LOCATION}/RELEASE.INFO" "${DESTINATION_RELEASE_LOCATION}"
+					cp "${CORTX_ISO_LOCATION}/RELEASE.INFO" .
+				'''		
+				archiveArtifacts artifacts: "RELEASE.INFO", onlyIfSuccessful: false, allowEmptyArchive: true
+			}
+		}
+		stage ("Clean Cluster and create image") {
+			parallel {
+				stage ("Cluster Cleanup") {
+					steps {
+						script { build_stage = env.STAGE_NAME }
+						script {
+							build job: '/Cortx-kubernetes/destroy-cortx-cluster', wait: true,
+							parameters: [
+								string(name: 'CORTX_RE_BRANCH', value: "${CORTX_RE_BRANCH}"),
+								string(name: 'CORTX_RE_REPO', value: "${CORTX_RE_REPO}"),
+								text(name: 'hosts', value: "${hosts}")
+							]
+						}
+					}
+				}
+				stage ("CORTX-ALL image creation") {
+					steps {
+						script { build_stage = env.STAGE_NAME }
+						script {
+							try {
+								def buildCortxDockerImages = build job: '/Cortx-kubernetes/cortx-all-docker-image', wait: true,
+								parameters: [
+									string(name: 'CORTX_RE_URL', value: "${CORTX_RE_REPO}"),
+									string(name: 'CORTX_RE_BRANCH', value: "${CORTX_RE_BRANCH}"),
+									string(name: 'BUILD', value: "${CORTX_BUILD}"),
+									string(name: 'EMAIL_RECIPIENTS', value: "${EMAIL_RECIPIENTS}"),
+									string(name: 'GITHUB_PUSH', value: "${PUSH_IMAGE}"),
+									string(name: 'TAG_LATEST', value: "${TAG_LATEST}"),
+									string(name: 'DOCKER_REGISTRY', value: "${DOCKER_REGISTRY}")
+								]
+								env.dockerimage_id = buildCortxDockerImages.buildVariables.image
+							} catch (err) {
+								build_stage = env.STAGE_NAME
+								error "Failed to Build Docker Image"
+							}
+						}
+					}
+				}
+			}
+		}
+		stage ("Deploy CORTX Cluster") {
```

Comment:
Can't we use groovy file used for below job and create one independent deployment job for PR-deployments only instead of triggering cleanup and cortx-deploy separately?

http://eos-jenkins.colo.seagate.com/job/Cortx-Kubernetes/job/post-merge/view/Kubernetes_Dashboard/job/K8s-3N-deployment/

CC - @shailesh-vaidya

Reply:
@gauravchaudhari02  - Let it be as it is for now. I am working on adding functionality in `setup-cortx-cluster` job itself to clean up first if the cluster is already deployed.